packages/eslint-plugin/src/rules/consistent-callback-type/index.test.js

```diff
@@ -0,0 +1,69 @@
+// @ts-nocheck
+const { ESLintUtils } = require('@typescript-eslint/experimental-utils');
+const rule = require('./index');
+
+const ruleTester = new ESLintUtils.RuleTester({
+  parser: '@typescript-eslint/parser',
+});
+
+ruleTester.run('consistent-callback-type', rule, {
+  valid: [
+    // Valid when prop is TSTypeAliasDeclaration and the callback uses EventHandler
+    {
+      code: `
+        import { EventHandler, EventData } from '@fluentui/react-utilities';
+        export type OnSomeEventData = EventData<'focus', React.FocusEvent<HTMLElement>> & {
+          open: boolean,
+        };
+        export type SomeProps = {
+          onSomeEvent?: EventHandler<OnSomeEventData>,
+        };
+      `,
+      filename: 'src/components/SomeComponent/SomeComponent.type.ts',
+    },
+    // Valid when prop is TSIntersectionType and the callback uses EventHandler
+    {
+      code: `
+        import { EventHandler, EventData } from '@fluentui/react-utilities';
+        export type OnSomeEventData = EventData<'focus', React.FocusEvent<HTMLElement>> & {
+          open: boolean,
+        };
+        export type SomeProps = SomeType & {
+          onSomeEvent?: EventHandler<OnSomeEventData>,
+        };
+      `,
+      filename: 'src/components/SomeComponent/SomeComponent.type.ts',
+    },
+    // Valid when prop has a function that is not callback, and not using EventHandler type
+    {
+      code: `
+        type SomeArgBag = { one: number };
+        export type SomeProps = {
+          someFunction?: (args: SomeArgBag) => void;
+          someFunction2?: (args: { onSomething: string }) => void; // test nested TSTypeLiteral node that starts with 'on'
+        };
+      `,
+      filename: 'src/components/SomeComponent/SomeComponent.type.ts',
+    },
+  ],
+  invalid: [
+    // Invalid when callback in props is not using EventHandler
+    {
+      code: `
+        export type SomeProps = {
+          onChange?: (ev: React.ChangeEvent<HTMLInputElement>, data: {}) => void;
+        };
+      `,
+      errors: [{ messageId: 'invalidType' }],
+    },
+    // Invalid when callback in props is not using EventHandler, and the props is TSIntersectionType
+    {
+      code: `
+        export type SomeProps = SomeType & {
```

Comment:
suggestion: WDYT about adding double literal with intersection to make the checks more robust

```suggestion
        export type SomeProps = SomeType & {onClick?: (ev: React.ClickEvent<HTMLInputElement>) => void;} {
```

Reply:
Added